src/mesh/FloodingRouter.cpp

```diff
@@ -29,6 +29,13 @@ bool FloodingRouter::shouldFilterReceived(const meshtastic_MeshPacket *p)
             // cancel rebroadcast of this message *if* there was already one, unless we're a router/repeater!
             if (Router::cancelSending(p->from, p->id))
                 txRelayCanceled++;
+        } else if (iface && p->hop_limit > 0) {
+            // If we overhear a duplicate copy of the packet with more hops left than the one we are waiting to
+            // rebroadcast, then remove the packet currently sitting in the TX queue and use this one instead.
+            if (iface->removePendingTXPacket(getFrom(p), p->id, p->hop_limit - 1)) {
```

Comment:
Note to self: need to also catch packets that arrived initially with `hop_limit` of `0`, and then subsequently with something higher.

As currently implemented, such packets will never hit the TX queue (because they weren't eligible for rebroadcast the first time around), and therefore will never be noticed as eligible for reprocessing.